src/x509.rs

```diff
@@ -239,9 +243,12 @@ impl<'a> FromDer<'a> for SubjectPublicKeyInfo<'a> {
                 }
             })(i)
             .or(Err(X509Error::InvalidSPKI))?;
+            let end = i;
+            let raw = &sequence[..2 + begin.len() - end.len()]; // 2 additional bytes for SEQUENCE
```

Comment:
This line looks clearly suspicious to me:
- the hardcoded `2` is wrong (the DER length could be on 1 byte if < 127, and eventually more than 2 bytes
- additionally, `begin` already points to the sequence *content* (not header), so length is not present
- the comment on the `raw` member seems to indicate that the header is included (otherwise, the result cannot be parsed as a DER object).

Anyway, this is not consistent with other `raw` fields (like in `TbsCertificate`), which include the header.
I'd rather stay consistent with other parsers. Could you update the patch, using the same method as for ex in https://github.com/rusticata/x509-parser/blob/x509-parser-0.9.x/src/certificate.rs#L244-L273 ? (include the header, and use the `start_i.offset(i)` method to obtain the length).
Thanks!

Reply:
I will try

Reply:
Done! It worked. If only I knew about `nom::traits::Offset`... =)